frame/system/src/lib.rs

```diff
@@ -1051,30 +1065,34 @@ impl<T: Config> Module<T> {
 		})
 	}
 
-	/// Decrement the reference counter on an account. This *MUST* only be done once for every time
-	/// you called `inc_consumers` on `who`.
+	/// Decrement the provider reference counter on an account.
+	///
+	/// This *MUST* only be done once for every time you called `inc_providers` on `who`.
 	pub fn dec_providers(who: &T::AccountId) -> Result<DecRefStatus, DecRefError> {
 		Account::<T>::try_mutate_exists(who, |maybe_account| {
 			if let Some(mut account) = maybe_account.take() {
-				match (account.providers, account.consumers) {
-					(0, _) => {
-						// Logic error - cannot decrement beyond zero and no item should
-						// exist with zero providers.
-						log::error!(
-							target: "runtime::system",
-							"Logic error: Unexpected underflow in reducing provider",
-						);
-						Ok(DecRefStatus::Reaped)
-					},
-					(1, 0) => {
+				if account.providers == 0 {
+					// Logic error - cannot decrement beyond zero.
+					log::error!(
+						target: "runtime::system",
+			      "Logic error: Unexpected underflow in reducing provider",
```

Comment:
```suggestion
						"Logic error: Unexpected underflow in reducing provider",
```

Do I see spaces here? :P

Reply:
Github's horrible merge tool :)